src/cnv/ugbio_cnv/convert_combined_cnv_results_to_vcf.py

```diff
@@ -0,0 +1,183 @@
+    )
+    header.add_line(
+        '##INFO=<ID=RoundedCopyNumber,Number=1,Type=Integer,Description="rounded copy number detected from cn.mops">'
+    )
+    header.add_line('##INFO=<ID=SVLEN,Number=.,Type=Integer,Description="CNV length">')
+    header.add_line('##INFO=<ID=SVTYPE,Number=1,Type=String,Description="CNV type. can be DUP or DEL">')
+    header.add_line(
+        '##INFO=<ID=CNV_SOURCE,Number=1,Type=String,Description="the tool called this CNV. \
+        can be combination of: cn.mops, cnvpytor, gridss">'
+    )
+
+    # Add FORMAT
+    header.add_line('##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">')
+
+    return header
+
+
+def write_vcf(outfile, header, cnv_annotated_bed_file, sample_name):
+    with pysam.VariantFile(outfile, mode="w", header=header) as vcf_out:
+        df_cnvs = pd.read_csv(cnv_annotated_bed_file, sep="\t", header=None)
+        df_cnvs.columns = ["chr", "start", "end", "CNV_type", "CNV_calls_source", "copy_number", "UG-CNV-LCR"]
+
+        for _, row in df_cnvs.iterrows():
+            # Create a new VCF record
+            chr_id = row["chr"]
+            start = row["start"]
+            end = row["end"]
+            cnv_type = row["CNV_type"]
+            cnv_call_source = row["CNV_calls_source"]
+            copy_number = row["copy_number"]
+            ug_cnv_lcr = row["UG-CNV-LCR"]
+
+            cn_list = copy_number.split(",")
+            cn_list_filtered = [float(item) for item in cn_list if item not in (["DUP", "DEL"])]
+
+            copy_number_value = cn_list[0]
+            if len(cn_list_filtered) > 0:
+                copy_number_value = statistics.mean(cn_list_filtered)
+
+            cnv_type_value = f"<{cnv_type}>"
+
+            ug_cnv_lcr_value = "UG-CNV-LCR" if ug_cnv_lcr != "." else ""
+
+            record = vcf_out.new_record()
+            record.contig = chr_id
+            record.start = start
+            record.stop = end
+            record.ref = "N"
+            record.alts = (cnv_type_value,)
+            if ug_cnv_lcr_value == "UG-CNV-LCR":
+                record.filter.add("UG-CNV-LCR")
+            else:
+                record.filter.add("PASS")
+            if isinstance(copy_number_value, float):
+                record.info["CopyNumber"] = copy_number_value
+            if isinstance(copy_number_value, float):
+                record.info["RoundedCopyNumber"] = int(round(copy_number_value))
+            record.info["SVLEN"] = int(end) - int(start)
+            record.info["SVTYPE"] = cnv_type
+            record.info["CNV_SOURCE"] = cnv_call_source
+            # END position is automatically generated for multi-base variants
+
+            # Set genotype information for each sample
+            gt = [None, 1]
+            if isinstance(copy_number_value, float):
+                if int(round(copy_number_value)) == 1:
+                    gt = [0, 1]
+                elif int(round(copy_number_value)) == 0:
+                    gt = [1, 1]
```

Comment:
What happens if it is more than 1? We just say nothing? Maybe it was a concise decision and I just don't remember. In any case there is a lot of duplication here with the flow of `convert_cnv_results_to_vcf`

Reply:
this is the GT values we decided for deletions as described in other VCFs representing CNVs. 
the copy_number_value is the average value of all reported copy numbers for that specific segments. we round it and hold it as a single value. 
I haven't seen a case for both DUP and DEL tag for a single segment. maybe an overlap. but in this case we will have a record per CNV.